### PR TITLE
passExtensions.pass-file: init at 1.0.0

### PR DESCRIPTION
**Pull request by taranarmo**
## Description of changes

Pass extension to save files inside its storage
https://github.com/dvogt23/pass-file
It is a bash file and subcommand of pass. I am new to the NixOS thus I wasn't able to connect the extension to pass to test it properly. Though makefile was provided by the author and I used it to build the package.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ghost — COMMENTED** [review]
a couple more nits i just learned from my own PR request.
[edit] unclear to me that it makes a difference so removed the inline comments.

though the larger issue as noted in the PR request

> I wasn't able to connect the extension to pass to test it properly. Though makefile was provided by the author and I used it to build the package.

seems like it needs to be addressed.

**Comment by robert-manchester** [comment]
> I wasn't able to connect the extension to pass to test it properly. Though makefile was provided by the author and I used it to build the package.

using `nix-shell` and pointing nixpkgs to the source root works for me.

```console
$ cd nixpkgs-srcroot
$ nix-shell -p 'pass.withExtensions(e: with e; [ pass-file ])' -I nixpkgs=`pwd`
$ source <(sed '/exec/d' `which pass`)
$ ls $SYSTEM_EXTENSION_DIR
file.bash
```

File: pkgs/tools/security/pass/extensions/file.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchFromGitHub, pass, shellcheck }:
```
Comment (robert-manchester) [inline pkgs/tools/security/pass/extensions/file.nix:1]:
```suggestion
{ lib, stdenv, fetchFromGitHub }:
```

File: pkgs/tools/security/pass/extensions/file.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "dvogt23";
+    repo = pname;
+    rev = version;
+    hash = "sha256-18KvmcfLwelyk9RV/IMaj6O/nkQEQz84eUEB/mRaKE4=";
+  };
+
+  nativeBuildInputs = [ shellcheck ];
```
Comment (robert-manchester) [inline pkgs/tools/security/pass/extensions/file.nix:14]:
```suggestion
```

**Review by ghost — COMMENTED** [review]
> I wasn't able to connect the extension to pass to test it properly. Though makefile was provided by the author and I used it to build the package.

using `nix-shell` and pointing nixpkgs to the source root works for me.

```console
$ cd nixpkgs-srcroot
$ nix-shell -p 'pass.withExtensions(e: with e; [ pass-file ])' -I nixpkgs=`pwd`
$ source <(sed '/exec/d' `which pass`)
$ ls $SYSTEM_EXTENSION_DIR
file.bash
```

was able to remove a couple of inputs as well.

**Comment by taranarmo** [comment]
@robert-manchester, thank you so much for your comments, I should learn more about Nix and NixOS (I use it only couple of months and this is my first experience of creating a package anywhere). Following your steps I was able to connect the extension to the main program, seems to work properly. I updated the commit according to your last suggestions.

**Review by ghost — APPROVED** [review]

**Comment by emilytrau** [comment]
Result of `nixpkgs-review pr 262046` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>passExtensions.pass-file</li>
  </ul>
</details>

**Review by emilytrau — APPROVED** [review]

**Comment by emilytrau** [comment]
Thanks for your contribution and welcome to nixpkgs!

**Pull request merged**